tests/TestCase/ORM/EntityTest.php

```diff
@@ -46,6 +46,31 @@ public function testSetOneParamNoSetters()
         $this->assertSame(1, $entity->id);
         $this->assertEquals(1, $entity->getOriginal('id'));
         $this->assertEquals('bar', $entity->getOriginal('foo'));
+
+        $entity->set('level1.level2.level3', 'three');
+        $this->assertSame('three', $entity->level1->level2->level3);
+        $this->assertEquals('three', $entity->getOriginal('level1.level2.level3'));
+        $entity->set('level1.level2.level3', '333');
+        $this->assertEquals('three', $entity->getOriginal('level1.level2.level3'));
```

Comment:
...Or are you saying we should start storing more fine grained dirty state for **elements** inside of dirty properties?